pkg/isb/stores/jetstream/writer.go

```diff
@@ -252,8 +260,34 @@ func (jw *jetStreamWriter) Write(_ context.Context, messages []isb.Message) ([]i
 		cancel()
 		<-futureCheckDone
 		// TODO: Maybe need to reconnect.
-		isbWriteTimeout.With(labels).Inc()
+		isbWriteTimeout.With(metricsLabels).Inc()
+	}
+	return writeOffsets, errs
+}
+
+func (jw *jetStreamWriter) syncWrite(_ context.Context, messages []isb.Message, errs []error, metricsLabels map[string]string) ([]isb.Offset, []error) {
+	var writeOffsets = make([]isb.Offset, len(messages))
+	wg := new(sync.WaitGroup)
+	for index, msg := range messages {
+		wg.Add(1)
+		go func(message isb.Message, idx int) {
+			defer wg.Done()
+			m := &nats.Msg{
+				Header:  convert2NatsMsgHeader(message.Header),
+				Subject: jw.subject,
+				Data:    message.Payload,
+			}
+			if pubAck, err := jw.js.PublishMsg(m, nats.MsgId(message.Header.ID)); err != nil { // nats.MsgId() is for exactly-once writing
```

Comment:
should we do a timeout to avoid a hanging ack?

Reply:
added `AckWait`